src/runtime/inline/list.h

```diff
@@ -48,12 +48,12 @@ extern "C" inline void listAppendInternalStolen(Box* s, Box* v) {
     assert(PyList_Check(s));
     BoxedList* self = static_cast<BoxedList*>(s);
 
-    assert(self->size <= self->capacity);
```

Comment:
I think we wan't to keep this assert but it should instead be `assert(self->size <= self->capacity || self->capacity == -1);`